### PR TITLE
docs(datasets) Add dataset contribution guide

### PR DESCRIPTION
**Pull request by adam-narozniak**

### CODE REVIEW

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst [hunk 1]
```diff
@@ -0,0 +1,57 @@
+How to contribute a dataset
+===========================
+
+To make a dataset available in Flower Dataset (`flwr-datasets`), you need to add the dataset to `HuggingFace Hub <https://huggingface.co/>`_ .
+
+This guide will explain the best practices we found when adding datasets ourselves and point to the HFs guides. To see the datasets added by Flower, visit https://huggingface.co/flwrlabs.
+
+Dataset contribution process
+----------------------------
+The contribution contains two steps:
```
Comment (jafermarq) [inline datasets/doc/source/contributor-how-to-contribute-dataset.rst:10]:
```suggestion
The contribution contains three steps: first, on your development machine transform your dataset into a ``datasets.Dataset`` object, the preferred format for datasets in HF Hub; second, upload the dataset to HuggingFace Hub and detail it its readme how can be used with Flower Dataset; third, share your dataset with us and we will add it to the `recommended FL dataset list <https://flower.ai/docs/datasets/recommended-fl-datasets.html>`_```

Comment (jafermarq) [reply datasets/doc/source/contributor-how-to-contribute-dataset.rst:10]: I'm thinking having a tiny paragraph and then jump straight in to the two subsections below is better than having a lists and then two sub-sections. Also, what do you think about framing the process as a three stage process ?

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+1. Create the dataset locally.
+
+We recommend that you do not upload custom scripts to HuggingFace Hub; instead, create the dataset locally and upload the data, which will speed up the processing time each time the data set is downloaded.
```
Comment (jafermarq) [inline datasets/doc/source/contributor-how-to-contribute-dataset.rst:14]: useful info, how about having it as a `:: tip:` right under the `Creating a datasets locally` section?

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+2. Contribute to HuggingFace Hub.
+
+Each dataset in the HF Hub is a Git repository with a specific structure and readme file, and HuggingFace provides an API to push the dataset and, alternatively, a user interface directly in the website to populate the information in the readme file.
```
Comment (jafermarq) [inline datasets/doc/source/contributor-how-to-contribute-dataset.rst:18]: I'd move this as a mini introduction to the `Contribution to the HuggingFace Hub` section. It's good

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst [hunk 4]
```diff
@@ -0,0 +1,57 @@
+How to contribute a dataset
+===========================
+
+To make a dataset available in Flower Dataset (`flwr-datasets`), you need to add the dataset to `HuggingFace Hub <https://huggingface.co/>`_ .
```
Comment (jafermarq) [inline datasets/doc/source/contributor-how-to-contribute-dataset.rst:4]:
```suggestion
To make a dataset available in Flower Dataset (`flwr-datasets`) for other in the community to use, it needs to be added to `HuggingFace Hub <https://huggingface.co/>`_ .
```

**Review by jafermarq — COMMENTED** [review]
Hey @adam-narozniak , it looks pretty good. I like the inclusion of a mini example at the bottom. Some comments below to move some text around a bit.

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst (hunk 1 above)
Comment (adam-narozniak) [reply datasets/doc/source/contributor-how-to-contribute-dataset.rst:10]: Ok, I'll change that

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst (hunk 4 above)
Comment (adam-narozniak) [reply datasets/doc/source/contributor-how-to-contribute-dataset.rst]: I went with my version for this

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
```
Comment (jafermarq) [inline datasets/doc/source/contributor-how-to-contribute-dataset.rst:4]:
```suggestion
To make a dataset available in Flower Dataset (`flwr-datasets`), you need to add the dataset to `HuggingFace Hub <https://huggingface.co/datasets>`_ .
```

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        'column2': ['a', 'b', 'c']
+    }
+
+    # Create a Dataset object
+    dataset = Dataset.from_dict(data)
+
+    # Push the dataset to the HuggingFace Hub
+    dataset.push_to_hub("you-hf-username/your-ds-name")
+
+To make the dataset easily accessible in FL we recommend adding the "Use in FL" section. Here's an example of how it is done in `one of our reps  <https://huggingface.co/datasets/flwrlabs/cinic10#use-in-fl>`_ for the cinic10 dataset.
```
Comment (jafermarq) [inline datasets/doc/source/contributor-how-to-contribute-dataset.rst:50]:
```suggestion
To make the dataset easily accessible in FL we recommend adding the "Use in FL" section. Here's an example of how it is done in `one of our repos  <https://huggingface.co/datasets/flwrlabs/cinic10#use-in-fl>`_ for the cinic10 dataset.
```

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+This guide will explain the best practices we found when adding datasets ourselves and point to the HFs guides. To see the datasets added by Flower, visit https://huggingface.co/flwrlabs.
+
+Dataset contribution process
+----------------------------
+The contribution contains three steps: first, on your development machine transform your dataset into a ``datasets.Dataset`` object, the preferred format for datasets in HF Hub; second, upload the dataset to HuggingFace Hub and detail it its readme how can be used with Flower Dataset; third, share your dataset with us and we will add it to the `recommended FL dataset list <https://flower.ai/docs/datasets/recommended-fl-datasets.html>`_ 
+
+Creating a dataset locally
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+You can create a local dataset directly using the `datasets` library or load it in any custom way and transform it to the `datasets.Dataset` from other Python objects.
+To complete this step, we recommend reading our guide available here: :doc:`how-to-use-with-local-data` or/and reading the guide from HF `Create a dataset <https://huggingface.co/docs/datasets/create_dataset>`_.
```
Comment (jafermarq) [inline datasets/doc/source/contributor-how-to-contribute-dataset.rst:15]:
```suggestion
To complete this step, we recommend reading our :doc:`how-to-use-with-local-data` guide or/and the `Create a dataset <https://huggingface.co/docs/datasets/create_dataset>`_ guide from HF.
```

Comment (jafermarq) [reply datasets/doc/source/contributor-how-to-contribute-dataset.rst:15]: just a minor rewrite so the paragraph doesn't have an extra line with just one word in it (i.e. gives the same info but in a bit less space)

File: datasets/doc/source/contributor-how-to-contribute-dataset.rst (hunk 5 above)
Comment (jafermarq) [reply datasets/doc/source/contributor-how-to-contribute-dataset.rst:4]: better this?

**Review by jafermarq — APPROVED** [review]

**Pull request merged**